src/components/views/rooms/Autocomplete.js

```diff
@@ -163,26 +163,17 @@ export default class Autocomplete extends React.Component {
     }
 
     // called from MessageComposerInput
-    onUpArrow(): ?Completion {
+    moveSelection(delta): ?Completion {
```

Comment:
Nice cleanup! Can you also (rebase your branch and) adjust `/src/editor/autocomplete.js` to also call this method instead?:
```
    onUpArrow() {
        this._getAutocompleterComponent().onUpArrow();
    }

    onDownArrow() {
        this._getAutocompleterComponent().onDownArrow();
    }
```

Reply:
Updated, as well as `countCompletions` for consistency.
I've guessed and removed the `onUpArrow` and `onDownArrow` methods here too, but that may not be what you want (I take it that as this seems to be some sort of wrapper, you might want to expose it elsewhere). Can be reverted if needed.

Reply:
Nevermind, I see where it fails now, `MessageEditor` was calling those functions. They've been added back, and I've also added `onLeftArrow`/`onRightArrow` to match.

Similarly, `MessageEditor` wasn't featuring left/right autocomplete navigation at all, so I made it call those two extra functions on `ArrowLeft` and `ArrowRight` events respectively.